src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1530,6 +1544,107 @@ def _get_linux_multicontainer_encoded_config_from_file(file_name):
+    is_flex = is_flex_functionapp(cmd.cli_ctx, resource_group_name, name)
+
+    if not is_flex:
+        raise ValidationError("This command is only valid for Azure Functions on the FlexConsumption plan.")
+
+    logger.warning("deployment_storage_name: %s deployment_storage_container_name: %s", deployment_storage_name, deployment_storage_container_name)
+    logger.warning("deployment_storage_auth_type: %s deployment_storage_auth_value: %s", deployment_storage_auth_type, deployment_storage_auth_value)
+
+    if (deployment_storage_name == None) == (deployment_storage_container_name != None):
+        raise ValidationError("Please provide both --deployment-storage-name and --deployment-storage-container-name or neither.")
+    
+    client = web_client_factory(cmd.cli_ctx)
+    functionapp = client.web_apps.get(resource_group_name, name)
+    
+    if functionapp is not None:
+        logger.warning("functionapp %s in %s found", functionapp.name, functionapp.location)
+
+    deployment_config_storage_value = None
+    if (deployment_storage_name is not None):
+        deployment_storage = _validate_and_get_deployment_storage(cmd.cli_ctx, resource_group_name, deployment_storage_name)
+        deployment_storage_container = _get_deployment_storage_container(cmd, resource_group_name, deployment_storage_name, deployment_storage_container_name)
+        deployment_storage_container_name = deployment_storage_container.name
+        deployment_config_storage_value = getattr(deployment_storage.primary_endpoints, 'blob') + deployment_storage_container_name
+    else:
+        # update_flex_functionapp_functions_configuration();
+        # todo: hard-coded for testing. Use existing deployment config. 
+        deployment_storage_name = 'khkhflexstorage'
+        deployment_storage = _validate_and_get_deployment_storage(cmd.cli_ctx, resource_group_name, deployment_storage_name)
+    
+    deployment_storage_auth_type = deployment_storage_auth_type or 'systemAssignedIdentity'
+
+    if deployment_storage_auth_value and deployment_storage_auth_type != 'userAssignedIdentity':
+        raise ArgumentUsageError(
+            '--deployment-storage-auth-value is only a valid input for --deployment-storage-auth-type set to userAssignedIdentity. '
+            'Please try again with --deployment-storage-auth-type set to userAssignedIdentity.'
+        )
+
+    if deployment_storage_auth_type == 'userAssignedIdentity':
+        deployment_storage_user_assigned_identity = _get_or_create_user_assigned_identity(cmd, resource_group_name, name, deployment_storage_auth_value, functionapp.location)
+        deployment_storage_auth_value = deployment_storage_user_assigned_identity.id
+    elif deployment_storage_auth_type == 'storageAccountConnectionString':
+        logger.warning("deployment_storage_auth_type is %s", deployment_storage_auth_type)
+        deployment_storage_conn_string = _get_storage_connection_string(cmd.cli_ctx, deployment_storage)
```

Comment:
My understanding is that customers were already going to pass the connection string, not that we would obtain it for them

Reply:
Do we ask connection string in any other command? I thought that we always only ask for storage account name and we always obtain the connection string for the customer.